### ISSUE TITLE
Raspberry Pi build fails

### ISSUE THREAD

**Issue by ryonakano** [elementary/os#727]
### What Happened?
The latest Raspberry Pi image build [fails](https://github.com/elementary/os/actions/runs/11578141147/job/32434307382) with the following error:

```
I: Retrieving xxd 2:9.1.0016-1ubuntu7
I: Validating xxd 2:9.1.0016-1ubuntu7
I: Retrieving zlib1g 1:1.3.dfsg-3.1ubuntu2
I: Validating zlib1g 1:1.3.dfsg-3.1ubuntu2
I: Chosen extractor for .deb packages: dpkg-deb
I: Extracting base-files...
E: Tried to extract package, but file already exists. Exit...
Error: Process completed with exit code 1.
```

I can reproduce this by building the image locally and according to the debootstrap.log this is the same issue with https://bugs.launchpad.net/ubuntu/+source/base-files/+bug/2054925

```
[ryo@b760m ~/work/elementary/os (ryonakano/rpi-fix-build $)]$ docker run --rm --privileged -it \
    -v /proc:/proc \
    -v ${PWD}:/working_dir \
    -w /working_dir \
    ubuntu:20.04
root@e9b961854ea7:/working_dir# tail -n5 artifacts/elementary-rpi/elementary-arm64/debootstrap/debootstrap.log
2024-11-03 06:22:46 URL:http://ports.ubuntu.com/ubuntu-ports/pool/main/z/zlib/zlib1g_1.3.dfsg-3.1ubuntu2_arm64.deb [61682/61682] -> "/working_dir/artifacts/elementary-rpi/elementary-arm64//var/cache/apt/archives/partial/zlib1g_1%3a1.3.dfsg-3.1ubuntu2_arm64.deb" [1]
tar: ./bin: Cannot create symlink to 'usr/bin': File exists
tar: ./lib: Cannot create symlink to 'usr/lib': File exists
tar: ./sbin: Cannot create symlink to 'usr/sbin': File exists
tar: Exiting with failure status due to previous errors
root@e9b961854ea7:/working_dir#
```

### Steps to Reproduce

- See the latest build log
- Alternatively try building the Raspberry Pi image locally

### Expected Behavior

The build succeeds.

**Comment by stsdc** [comment]
Is this for RaspberryPi or Pinebook? Because I see `ubuntu:20.04` in your command and it is part of Pinebook command?

According the readme this is the right command for RaspberryPi:
```
docker run --rm --privileged -it \
    -v /proc:/proc \
    -v ${PWD}:/working_dir \
    -w /working_dir \
    ubuntu:22.04 \
    ./build-rpi.sh
```

but for pinebook it's still 20.04:
```
docker run --rm --privileged -it \
    -v /proc:/proc \
    -v ${PWD}:/working_dir \
    -w /working_dir \
    ubuntu:20.04 \
    ./build-pinebookpro.sh
```

**Comment by stsdc** [comment]
I can see in changelogs of debootstrap there is an attempt to fix symlinks issue.
Can you try a newer version of debootstrap? 
http://archive.ubuntu.com/ubuntu/pool/main/d/debootstrap/debootstrap_1.0.137ubuntu3_all.deb

For me it not much helped:
```
stsdc@desktop:~/projects/os$ tail -n15 artifacts/elementary-rpi/elementary-arm64/debootstrap/debootstrap.log
tar: ./usr/share/pixmaps/ubuntu-logo-text-dark.png: Cannot open: File exists
tar: ./usr/share/pixmaps/ubuntu-logo-text.png: Cannot open: File exists
tar: ./usr/share/pixmaps/ubuntu-logo.svg: Cannot open: File exists
tar: ./usr/share/profile: Cannot open: File exists
tar: ./usr/share/profile.md5sums: Cannot open: File exists
tar: ./usr/share/staff-group-for-usr-local: Cannot open: File exists
tar: ./bin: Cannot create symlink to 'usr/bin': File exists
tar: ./etc/os-release: Cannot open: File exists
tar: ./lib: Cannot create symlink to 'usr/lib': File exists
tar: ./sbin: Cannot create symlink to 'usr/sbin': File exists
tar: ./usr/share/common-licenses/GFDL: Cannot create symlink to 'GFDL-1.3': File exists
tar: ./usr/share/common-licenses/GPL: Cannot create symlink to 'GPL-3': File exists
tar: ./usr/share/common-licenses/LGPL: Cannot create symlink to 'LGPL-3': File exists
tar: ./usr/share/doc/base-files/FAQ: Cannot create symlink to 'README': File exists
tar: Exiting with failure status due to previous errors
```

**Comment by ryonakano** [comment]
Thank you for your info, I was on Ubuntu 20.04 when I filled this issue. Now I tested again on OS 8 RC and confirmed base-files extracted as expected (OS 8 RC comes with `debootstrap 1.0.134ubuntu1` but I guess it contains the fix for this issue).

The CI for image build currently runs on Ubuntu 20.04, so switching to Ubuntu 24.04 would fix the CI. I've started working on this here:

https://github.com/elementary/os/tree/ryonakano/arm64-build-on-noble

> For me it not much helped:

Did you possibly run the script without removing the artifacts from previous builds? Does it still occurs even if you first clean it and then run the script?

```
cd ~/projects/os
sudo rm -rf ./artifacts/elementary-rpi
sudo ./build-rpi.sh
```
